app/assets/javascripts/controllers/vm_cloud/vm_cloud_remove_security_group_form_controller.js

```diff
@@ -0,0 +1,40 @@
+ManageIQ.angular.app.controller('vmCloudRemoveSecurityGroupFormController', ['$http', 'vmCloudRemoveSecurityGroupFormId', 'miqService', function($http, vmCloudRemoveSecurityGroupFormId, miqService) {
```

Comment:
We are trying to convert to Angular 1.6 so that we can move forward with Angular 2 and React. This should be a component. The partials below shoud not be created, those should be part of the component or separate componets too.

Ping @himdel, @tzumainn. http://www.codelord.net/2015/10/07/angular-2-preparation-killing-controllers/

Reply:
There are **no controllers in Angular 2+**, so i completely that we should try to write any new angular **controllers** as **components**, since we will have to refactor them anyway... And that applies especially for partials, because refactoring them later after they go through several modifications is pain in the ass.

You can use [this article](https://teropa.info/blog/2015/10/18/refactoring-angular-apps-to-components.html) as reference. There are also some components already in ui-classic(and more is comming).

Reply:
Working on the guide, but for now, pretty much all you need to do is:

* rename `init` to `$onInit` - that gets called automagically by angular, so you no longer need to call it manually.
* move the templates under `app/views/static/` - that way, you can reference it via `templateUrl: '/static/....'` from angular
* remove any controller-dependent logic from the moved template - that just means you can't use `@record.id` in your case
* make it into a component...

```js
ManageIQ.angular.app.controller("foo", **controller**);
```

should become

```js
ManageIQ.angular.app.component("foo", {
  bindings: {
    recordId: "@",
  }
  templateUrl: "/static/foo.html.haml",
  controller: **controller**,
});
```

* then you no longer need to pass the id using `value`, you can provide it to the component as a parameter - hence the `bindings` entry - so no injecting `vmCloudRemoveSecurityGroupFormId` any more ;)
* replace the original template with just something that uses the component .. so, `%my-component{:record-id => @record.id}`, and initializes angular (so that `miq_bootstrap` stays here, not in `static` - we usually use the component name as the first arg in those cases (so you don't need an extra `#div`))
* don't leave the code in `app/assets/javascripts/controllers`, use the `components` dir for that - or if you're up to speed with the recent webpacker changes ([talk](http://talk.manageiq.org/t/webpack-er-for-ui/2605)), you can move it to `app/javascript/` and use newer JS features.

Reply:
Also, that `miqAjaxButton` is an anti-pattern now, with angular components, you should no longer rely on the server generating javascript to redirect you/show a flash message/whatever... instead, you should be using `render :json => ...` in ruby, and read that json in JS to do the right thing.

(But, it's not `miqAjaxButton(url, true)` so.. if the server-side logic to do that is not trivial, feel free to ignore this bit *for now*.)